src/lsm/binary_search.zig

```diff
@@ -43,9 +42,9 @@ pub fn binary_search_values_upsert_index(
     while (length > 1) {
         if (constants.verify) {
             assert(offset == 0 or
-                compare_keys(key_from_value(&values[offset - 1]), key) != .gt);
+                !(key_from_value(&values[offset - 1]) > key));
```

Comment:
It's an excellent opportunity to put https://github.com/tigerbeetle/tigerbeetle/pull/1179#discussion_r1322873550 in practice, any thoughts?

| before | after |
--------|--------
`compare_keys(a,b) == .eq` | `a == b`
`compare_keys(a,b) == .lt` | `a < b`
`compare_keys(a,b) == .gt` | `a > b`
`compare_keys(a,b) != .eq` | `a != b`
`compare_keys(a,b) != .lt` | `!(a < b)`
`compare_keys(a,b) != .gt` | `!(a > b)`

In some cases, especially in asserts, `!(a > b)` reads better than `a <= b` imo.

Reply:
>What is the expected semantics of maybe(<condition>)?

The intended semantics is that the condition can be either true or false, and, once we have branch coverage, we should have tests tripping both sides.

I don't think we assign any  probabilities to maybe, so both `maybe(cond)` and `maybe(!cond)` are equivalent. Though, I think we usually state the more surprising thing in the `cond`, as a cognitive help for the reader.